### PR TITLE
Fixed Issue #19

### LINKED ISSUE #19: Fix the bot!
So, I recently, carelessly merged #17 into the source of my Twitter bot, [TinyFlowerBeds](https://twitter.com/tinyflowerbeds). Something went completely wrong with `configparser`, however.

Here's what you'll need to do!
- Get the bot to read from a configuration file!
- Prove (with a screenshot or a log) that the bot can now parse the credentials properly!

### PR DESCRIPTION
**Pull request by oxi-git**
- Script now works with configparser. 
- README.md has been updated.

### CODE REVIEW

**Comment by n0toose** [comment]
Hey there, thanks for the contribution! Could you please squash the commits and then force-push? That'd be immensely appreciated, thanks!

Also, please submit changes irrelevant to fixing the aforementioned changes (e.g. changes to the README.md file) in a different pull request. (Please remove the Contributors section while you're at it, great work btw!)

File: src/bot.py
Comment (n0toose) [inline src/bot.py:88]: Have you really tested the bot? There are a few typos left in the source code that break it upon running.

File: src/bot.py
Comment (n0toose) [inline src/bot.py:119]: Could you document this change a bit better?

File: src/bot.py
Comment (n0toose) [inline src/bot.py:42]: Hm, what if the end user is running the bot manually, while having `cd'd` into the `src` directory itself?

**Review by n0toose — CHANGES_REQUESTED** [review]
Before merging, I'd like to ask you to please prove that the bot's working by providing one of the following:
- a screenshot on a mock Twitter account.
- proof that the config is being parsed properly (parsing values and printing inside of a terminal should be alright) and printed output of the `tweet` variable.

File: src/bot.py
Comment (n0toose) [reply src/bot.py:119]: Adding a comment above that briefly explains the logic of this in less than 80 columns (characters) per line should be alright.

If that's too big of a problem or if what I'm asking is too vague, I'll do it for you.

Comment (oxi-git) [reply src/bot.py:119]: I made the loop break after 16 times because **batch = batch[:16]** would keep only the first 16th items of that list anyway.

File: src/bot.py
Comment (oxi-git) [reply src/bot.py:88]: Where's the typo?

Comment (n0toose) [reply src/bot.py:88]: Isn't there an indentation error right there? Maybe I'm wrong, but this really doesn't feel right from my end.

image

Comment (oxi-git) [reply src/bot.py:88]: image
I am not seeing it on VSCode.

Comment (n0toose) [reply src/bot.py:88]: I'll confirm that in a bit, apologies.

File: README.md
Comment (n0toose) [inline README.md:48]: Please keep all README-related changes separate (as in in a different pull request, since updating the README is not relevant to resolving #19). You're also proposing to change the structure of the configuration file in a way that I'd accept-- please include that change in your changes.

**Review by n0toose — CHANGES_REQUESTED** [review]

File: src/bot.py
Comment (n0toose) [reply src/bot.py:119]: What'd happen if I randomly decided to manipulate the following configurations?

```
lines = 4
limit_per_line = 4
```

Comment (oxi-git) [reply src/bot.py:119]: It would still do it 16 times and output a max 4x4 flower square as before.

File: src/bot.py
Comment (oxi-git) [reply src/bot.py:42]: Simply wouldn't work. I just followed the original project structure.

Comment (n0toose) [reply src/bot.py:42]: Oh yeah, you're right about the fact that this is how I did it in the first place.

Please excuse me for those questions, it's just that although I've turned this bot into an educational project a while ago, the fact still is that I haven't touched it myself in a while and have forgotten a bunch of things revolving around it. 😬

File: src/bot.py
Comment (n0toose) [reply src/bot.py:119]: So, that line implicitly **DOES** impose a maximum limit of 4x4 as I'm suspecting, right?

Comment (oxi-git) [reply src/bot.py:119]: Yes, as batch = batch[:16] did. I just re wrote it in a much simple way.

Comment (n0toose) [reply src/bot.py:119]: You're right! Thanks for staying true to the original implementation, despite that being a temporary hack :P

**Comment by oxi-git** [comment]
> Before merging, I'd like to ask you to please prove that the bot's working by providing one of the following:
> 
> * a screenshot on a mock Twitter account.
> * proof that the config is being parsed properly (parsing values and printing inside of a terminal should be alright) and printed output of the `tweet` variable.

**Posting some proof of work**

proof1
proof2

**Comment by n0toose** [comment]
Understood, I'll merge this immediately after the undoing of the README.md changes.

Thanks for your resilience ;D

**Pull request merged**
